packages/message-manager/src/DecryptMessageManager.ts

```diff
@@ -130,6 +130,7 @@ export class DecryptMessageManager extends AbstractMessageManager<
     req?: OriginalRequest,
   ) {
     if (req) {
+      messageParams.requestId = req.id;
```

Comment:
Rather than doing this in every message manager, could we create a new method in the `AbstractMessageManager` such as `createUnapprovedMessage` that does all the overlapping stuff such as setting the `requestId` and the `id` and `time` etc?

Reply:
I think it could be a good time refactor and abstract common logic. Working on this now

Reply:
Updated [41878c9](https://github.com/MetaMask/core/pull/4636/commits/41878c93051745eaba1e927d53d954f1d0efbcec) 

let me know if there're any other suggestions!